jetstream_pt/engine.py

```diff
@@ -570,7 +584,19 @@ def generate(
     )
 
     next_token = self._sampling(logits, self.env.batch_size)
-    lens = decode_state.lens + 1
+    if self.env.ring_buffer:
+      input_pos = decode_state.input_pos + 1
+      lens = decode_state.lens + 1
+    else:
+      input_pos = jnp.where(
+          decode_state.input_pos == 0,
+          0,
+          decode_state.input_pos + 1 % self.env.cache_len,
+      )
+      lens = jnp.where(
+          decode_state.lens == 0, 0, decode_state.lens + 1 % self.env.cache_len
+      )
```

Comment:
If we delete the lens, any impact in non ring buffer case? My understanding there is 0 impact, correct me if i'm wrong.

Reply:
sounds good, thanks for more insights.